File: go/2017/day_1/day_1.go
package main

import "fmt"
import "strconv"

func main() {
        integerSequence := 1122
        var res int
        res = captchaSequenceSolver(integerSequence)
        fmt.Printf("%v", res)
}

func captchaSequenceSolver(integerSequence int) int {
        stringified := strconv.Itoa(integerSequence)
        firstChar := string(stringified[0])
        extended := stringified + firstChar

        var priorEl int
        var sumOfSequences int
        for i := 0; i < len(extended); i++ {
                el, _ := strconv.Atoi(string(extended[i]))
                if el == priorEl {
                        sumOfSequences += el
                }
                priorEl = el
        }
        return sumOfSequences

}

File: python/2017/day_1/day_1.py
def captcha_sequence_solver(int_sequence):
  listified = list(str(int_sequence))
  listified.append(listified[0])
  prior_el = ''
  sum_of_sequences = 0
  for el in listified:
    if el == prior_el:
      sum_of_sequences += int(el)
    prior_el = el
  return sum_of_sequences
